src/twisted/conch/scripts/ckeygen.py

```diff
@@ -202,45 +203,64 @@ def _defaultPrivateKeySubtype(keyType):
         return "PEM"
 
 
-def printFingerprint(options):
-    if not options["filename"]:
+def _getKeyOrDefault(options, input_collector=None):
+    """
+    If C{options["filename"]} is None, prompt the user to enter a path
+    or attempt to set it to .ssh/id_rsa
+    """
+    if input_collector is None:
+        input_collector = input
+    filename = options["filename"]
+    if not filename:
         filename = os.path.expanduser("~/.ssh/id_rsa")
-        options["filename"] = input("Enter file in which the key is (%s): " % filename)
-    if os.path.exists(options["filename"] + ".pub"):
-        options["filename"] += ".pub"
+        if platform.system() == "Windows":
+            filename = os.path.expandvars(R"%HOMEPATH %\.ssh\id_rsa")
+        filename = (
+            input_collector("Enter file in which the key is (%s): " % filename)
+            or filename
+        )
+    return filename
```

Comment:
Second half of the suggestion since github has some weird requirements on suggestion boundaries:

```suggestion
        if platform.system() == "Windows":
            filename = os.path.expandvars(R"%HOMEPATH %\.ssh\id_rsa")
        filename = (
            inputCollector("Enter file in which the key is (%s): " % filename)
            or filename
        )
    return filename
```

Reply:
Changed to camelCase.